wcomponents-theme/src/main/js/wc/file/util.js

```diff
@@ -0,0 +1,97 @@
+define(["wc/dom/uid", "wc/file/getMimeType"], function(uid, getMimeType) {
+	"use strict";
+	var instance = new FileUtil();
+	/**
+	 * Used when checking the newly created image is named with the correct extension.
+	 * If it does not already match any in the array then the extension at index zero will be appended.
+	 **/
+	FileUtil.prototype.mimeToExt = {
+		"image/jpeg": ["jpeg", "jpg"],
+		"image/png": ["png"],
+		"image/webp": ["webp"]
+	};
+
+	function FileUtil() {
+		/**
+		 * Converts a generic binary blob to a File blob.
+		 * @param {Blob} The binary blob.
+		 * @param {Object} [config] Attempt to set some of the file properties such as "type", "name"
+		 * @returns {File} The File blob.
+		 */
+		this.blobToFile = function (blob, config) {
+			var name,
+				filePropertyBag = {
+					type: blob.type,
+					lastModified: new Date()
+				};
+			if (config) {
+				if (!filePropertyBag.type) {
+					filePropertyBag.type = config.type;
+				}
+				name = config.name;
+			}
+			name = name || uid();
+
+			if (!blob.type) {
+				// noinspection JSAnnotator
+				blob.type = filePropertyBag.type;
+			}
+			blob.lastModifiedDate = filePropertyBag.lastModified;
+			blob.lastModified = filePropertyBag.lastModified.getTime();
+			blob.name = name;
+			checkFileExtension(blob);
+			return blob;
+		};
+		
+
+		/**
+		 * Ensures that the file name ends with an extension that matches its mime type.
+		 * @param file The file to check.
+		 */
+		function checkFileExtension(file) {
```

Comment:
Do you think this has some generic utility? Should we make this public so it can be reused by other modules?